code/src/pages/orders/OrdersPage.tsx

```diff
@@ -5,13 +5,18 @@ import OrdersList from "@/containers/orders-list/OrdersList";
 import AppBox from "@/components/app-box/AppBox";
 import AppTypography from "@/components/app-typography/AppTypography";
 
-import { useGetOrdersQuery } from "@/store/api/ordersApi";
+import { useGetUserOrdersQuery } from "@/store/api/ordersApi";
+import { useUserDetailsSelector } from "@/store/slices/userSlice";
 
 import "@/pages/orders/OrdersPage.scss";
 
 const OrdersPage = () => {
-  //@TODO: use id from slice
-  const { data: orderResponse, isLoading } = useGetOrdersQuery({ id: "123" });
+  const userDetails = useUserDetailsSelector();
+
+  const { data: orderResponse, isLoading } = useGetUserOrdersQuery({
+    // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
+    userId: userDetails?.id! //we are sure that userDetails is not null because we are using this page inside protected route
```

Comment:
We should create selector that cannot be return null for user details. I can suggest at least two approaches:

The first one:
```jsx
export const useUserDetailsSelector = () =>
  useAppSelector((store) => {
    const userDetails = store.user.userDetails;

    if (userDetails === null) {
      throw new Error("Unauthorized");
    }

    return userDetails;
  });
```

The second one:
```jsx
  export const useUserDetailsSelector = () =>
  useAppSelector((store) => {
    const userDetails = store.user.userDetails;

    if (userDetails !== null) {
      return userDetails;
    }

    return null as never;
  });
```

Personally first approach looks more logical to me because we should not ever access user details when user is unauthorized. It is like a context in this case, if we are outside the context, we should receive and error. Same here, if we are outside protected layout, we should get an error

Then we can remove `useUserRoleSelector` and use `useUserDetailsSelector` everywhere

Reply:
I'll create issue for that